pandas/tests/frame/test_constructors.py

```diff
@@ -1256,13 +1283,13 @@ def test_constructor_column_duplicates(self):
 
         tm.assert_frame_equal(df, edf)
 
-        idf = DataFrame.from_items(
-            [('a', [8]), ('a', [5])], columns=['a', 'a'])
+        idf = DataFrame.from_records([(8, 5)],
```

Comment:
It seems that some of the dataframe constructors don't allow duplicated columns while others do so I had to change the ``from_items`` in this test to ``from_records`` and ``from_dict(OrderedDict())`` to get the test passing. 

All of the other tests in this file using ``from_items`` are directly testing ``from_items`` so I left the check for the deprecation warning rather than trying to replace ``from_items`` with a different constructor.